tests/infra/clients.py

```diff
@@ -366,33 +367,50 @@ def request(self, request, is_signed=False):
 
 class WSClient:
     def __init__(
-        self,
-        host,
-        port,
-        cert,
-        key,
-        ca,
-        connection_timeout,
-        request_timeout,
-        *args,
-        **kwargs,
+        self, host, port, cert, key, ca, request_timeout, *args, **kwargs,
     ):
         self.host = host
         self.port = port
         self.cert = cert
         self.key = key
         self.ca = ca
         self.request_timeout = request_timeout
+        self.ws = None
 
-    def request(self, request):
-        # pylint: disable=unused-variable
-        ws = create_connection(
-            f"wss://{self.host}:{self.port}",
-            sslopt={"certfile": self.cert, "keyfile": self.key, "ca_certs": self.ca},
+    def request(self, request, is_signed=False):
+        assert not is_signed
+
+        if not self.ws:
+            LOG.info("Creating WSS connection")
+            self.ws = websocket.create_connection(
```

Comment:
We should raise a `CCFConnectionException` exception here is the connection fails.

Reply:
My understanding is that this is done at the level above by CCFClient, but I'll check.

Reply:
The underlying client implementation has to raise a `CCFConnectionException` is the initial connection fails.

Reply:
Ok, I see, I'll add that and make sure that's chained.